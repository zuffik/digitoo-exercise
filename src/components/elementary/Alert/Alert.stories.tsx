import * as React from 'react';
import {Alert} from './Alert';

export default {
    title: 'Elementary/Alert',
};

export const danger = () => <Alert type="danger">Alert text</Alert>;
export const success = () => <Alert type="success">Alert text</Alert>;
export const warning = () => <Alert type="warning">Alert text</Alert>;
