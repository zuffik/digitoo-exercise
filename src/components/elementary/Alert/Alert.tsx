import * as React from 'react';

interface Props {
  type: 'danger' | 'success' | 'warning';
  children: React.ReactNode;
}

export const Alert: React.FC<Props> = (props: Props) => {
  return <div className={`alert alert-${props.type}`}>{props.children}</div>;
};
