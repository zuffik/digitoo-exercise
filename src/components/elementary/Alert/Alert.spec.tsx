import * as React from 'react';
import {render} from '@testing-library/react'
import {Alert} from "./Alert";

describe('<Alert/>', () => {
    it('should have alert classes', () => {
        const {asFragment, getByText} = render(<Alert type="danger">alert content</Alert>);
        expect(getByText('alert content')).toHaveClass('alert alert-danger');
        expect(asFragment()).toMatchSnapshot();
    });
});
