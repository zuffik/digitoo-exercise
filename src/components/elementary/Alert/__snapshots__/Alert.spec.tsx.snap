// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert/> should have alert classes 1`] = `
<DocumentFragment>
  <div
    class="alert alert-danger"
  >
    alert content
  </div>
</DocumentFragment>
`;
